network/manager.go

```diff
@@ -681,3 +667,31 @@ func (nm *networkManager) GetEndpointID(containerID, ifName string) string {
 	}
 	return containerID + "-" + ifName
 }
+
+func cnsEndpointInfotoCNIEpInfo(endpointInfo restserver.EndpointInfo, endpointID string) *EndpointInfo {
+	epInfo := &EndpointInfo{
+		Id:                 endpointID,
+		IfIndex:            EndpointIfIndex, // Azure CNI supports only one interface
+		ContainerID:        endpointID,
+		PODName:            endpointInfo.PodName,
+		PODNameSpace:       endpointInfo.PodNamespace,
+		NetworkContainerID: endpointID,
+	}
+	for ifName, ipInfo := range endpointInfo.IfnameToIPMap {
+		epInfo.IPAddresses = ipInfo.IPv4
+		epInfo.IPAddresses = append(epInfo.IPAddresses, ipInfo.IPv6...)
+		epInfo.IfName = ifName
+		epInfo.HostIfName = ipInfo.HostVethName
+		epInfo.HNSEndpointID = ipInfo.HnsEndpointID
+	}
+	return epInfo
+}
+
+func validateEndpointResponse(endpointInfo restserver.EndpointInfo) (bool, error) {
+	for _, ipInfo := range endpointInfo.IfnameToIPMap {
+		if ipInfo.HnsEndpointID == "" && ipInfo.HostVethName == "" {
+			return false, errors.New("Get endpoint API returend with empty HNSEndpointID and HostVethName")
```

Comment:
Can we add the endpoint id and more info in the error message for which `HSNendpointid` and `Hostvethname` is not present ? 
Also we can the log here the data of the endpoint, pod name etc

Reply:
the error has been removed